### PR TITLE
More mind stuff

### PR DESCRIPTION
**Pull request by Bod9001**
moves some stuff to mind ,
spawns in brain separate from body
and add possessing stuff

### CODE REVIEW

File: UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs [hunk 1]
```diff
@@ -826,6 +826,22 @@ private void SetConsciousState(ConsciousState NewConsciousState)
...
+		{
+			foreach (var bodyPart in SurfaceBodyParts.ToArray())
+			{
+				if (bodyPart.BodyPartType == bodyPartAim)
+				{
+					return bodyPart;
+				}
+			}
+
+			return null;
```
Comment (Alainx277) [inline UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]:
```suggestion
			return SurfaceBodyParts.FirstOrDefault(b => b.BodyPartType == bodyPartAim);
```

File: UnityProject/Assets/Scripts/Managers/PlayerInfo.cs [hunk 2]
```diff
@@ -142,6 +144,11 @@ private void TryChangeName(string playerName)
 		name = uniqueName;
 	}
 
+	public void SetMind(Mind InMind)
```
Comment (Alainx277) [inline UnityProject/Assets/Scripts/Managers/PlayerInfo.cs:147]: Why not make the setter public?

File: UnityProject/Assets/Scripts/Player/IBrain.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	LivingHealthMasterBase LHMB { get; set; }
+
+	virtual LivingHealthMasterBase LivingHealthMasterBase()
+	{
+		return LHMB;
+	}
+
+
+
+
```
Comment (Alainx277) [inline UnityProject/Assets/Scripts/Player/IBrain.cs]:
```suggestion
```

File: UnityProject/Assets/Scripts/Player/IPlayerPossessble.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+using System.Collections;
+using System.Collections.Generic;
+using HealthV2;
+using UnityEngine;
+
+public interface IPlayerPossessble
```
Comment (Alainx277) [inline UnityProject/Assets/Scripts/Player/IPlayerPossessble.cs]:
```suggestion
public interface IPlayerPossessable
```

File: UnityProject/Assets/Scripts/UI/Systems/Lobby/CharacterSettings.cs [hunk 5]
```diff
@@ -264,6 +264,19 @@ public Gender GetGender()
...
+	{
+		foreach (var Race in RaceSOSingleton.Instance.Races)
+		{
+			if (Race.name == this.Species)
+			{
+				return Race;
+			}
+		}
+
+		return null;
```
Comment (Alainx277) [inline UnityProject/Assets/Scripts/UI/Systems/Lobby/CharacterSettings.cs]:
```suggestion
		return RaceSOSingleton.Instance.Races.FirstOrDefault(x => x.name == Species);
```

File: UnityProject/Assets/Scripts/Player/Mind.cs [hunk 6]
```diff
@@ -19,7 +19,14 @@
 /// </summary>
 public class Mind : MonoBehaviour
 {
+	public GameObject PossessingObject;
+	public IPlayerPossessble IPlayerPossessble;
```
Comment (Alainx277) [inline UnityProject/Assets/Scripts/Player/Mind.cs]:
```suggestion
	private GameObject possessedObject;
	private IPlayerPossessble possessed;
```

**Review by Alainx277 — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs]: like linq is nice but it's a pain to debug so idk

File: UnityProject/Assets/Scripts/Player/Mind.cs (hunk 6 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Player/Mind.cs]: I think you'd want these being public so you can access them through the mind, 

For example if you want to know where the player is you would do mind.possessedObject.position?

Comment (corp-0) [reply UnityProject/Assets/Scripts/Player/Mind.cs]: Public getter private setter?

File: UnityProject/Assets/Scripts/Player/IBrain.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+using System.Collections.Generic;
+using HealthV2;
+using UnityEngine;
+
+public interface IBrain
+{
+
+	LivingHealthMasterBase LHMB { get; set; }
+
+	virtual LivingHealthMasterBase LivingHealthMasterBase()
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Player/IBrain.cs]: I don't get the use of this method. You can get it from the property and implementors of the interface can modify the property as they please as well

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Player/IBrain.cs]: was just something to put into it is all wip
, and chunking the PR into smaller things

File: UnityProject/Assets/Scripts/Player/Mind.cs (hunk 6 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Player/Mind.cs]: ok

File: UnityProject/Assets/Scripts/Managers/PlayerInfo.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/PlayerInfo.cs:147]: more implicit way of setting the mind, so something is a bit more big than just setting A variable

**Pull request merged**